BuyPolish/Pola/UI/ProductSearch/ReportProblem/ReportProblemView.swift

```diff
@@ -103,42 +104,79 @@ final class ReportProblemView: UIView {
             height: helpLabel.height(forWidth: widthWithoutPadding)
         )
 
-        photoTitleLable.frameOrigin = CGPoint(
-            x: padding,
-            y: helpLabel.frame.maxY + verticalMargin
-        )
-
-        imagesContainer.frameOrigin = CGPoint(
-            x: padding,
-            y: photoTitleLable.frame.maxY + titleMargin
-        )
-        imagesContainer.frameSize = imagesContainer.sizeThatFits(CGSize(width: widthWithoutPadding, height: .zero))
-
-        descriptionTitleLabel.frameOrigin = CGPoint(
-            x: padding,
-            y: max(imagesContainer.frame.maxY + verticalMargin - bottomMargin, padding + topSafeAreaInset)
-        )
-
-        sendButtom.frame = CGRect(
-            x: padding,
-            y: bounds.height - padding - sendButtomHeight - bottomMargin,
-            width: widthWithoutPadding,
-            height: sendButtomHeight
-        )
-
-        descriptionTextView.frame = CGRect(
-            x: padding,
-            y: descriptionTitleLabel.frame.maxY + titleMargin,
-            width: widthWithoutPadding,
-            height: sendButtom.frame.minY - verticalMargin - descriptionTitleLabel.frame.maxY
-        )
+        if isImageContainerHidden {
+            descriptionTitleLabel.frameOrigin = CGPoint(
+                x: padding,
+                y: helpLabel.frame.maxY + verticalMargin
+            )
+
+            sendButtom.frame = CGRect(
+                x: padding,
+                y: bounds.height - padding - sendButtomHeight - bottomMargin,
+                width: widthWithoutPadding,
+                height: sendButtomHeight
+            )
+
+            descriptionTextView.frame = CGRect(
+                x: padding,
+                y: descriptionTitleLabel.frame.maxY + titleMargin,
+                width: widthWithoutPadding,
+                height: sendButtom.frame.minY - verticalMargin - descriptionTitleLabel.frame.maxY
+            )
+
+            descriptionBottomShadowView.frame = CGRect(
+                x: descriptionTextView.frame.minX,
+                y: descriptionTextView.frame.maxY,
+                width: descriptionTextView.frame.width,
+                height: descriptionShadowHeight
+            )
+        } else {
+            photoTitleLable.frameOrigin = CGPoint(
```

Comment:
Ten if jest strasznie duży. Można by to ładnie skrócić do:
```
        var descriptionTitleLabelY;
        if isImageContainerHidden == false {
            photoTitleLable.frameOrigin = CGPoint(
                x: padding,
                y: helpLabel.frame.maxY + verticalMargin
            )

            imagesContainer.frameOrigin = CGPoint(
                x: padding,
                y: photoTitleLable.frame.maxY + titleMargin
            )
            imagesContainer.frameSize = imagesContainer.sizeThatFits(CGSize(width: widthWithoutPadding, height: .zero))
          descriptionTitleLabelY = max(imagesContainer.frame.maxY + verticalMargin - bottomMargin, padding + topSafeAreaInset)
      } else {
        descriptionTitleLabelY = helpLabel.frame.maxY + verticalMargin
      }
     descriptionTitleLabel.frameOrigin = CGPoint(
           x: padding,
           y: descriptionTitleLabelY
)
```
Nie chcesz może przepisać tego na AutoLayout? W najgorszym przypadku będziesz musiał przepisać jeszcze 2 małe widoki:
 * `ReportImageView` 
 * `ReportImagesContainerView` - tu całą robotę by zrobił `UIStackView`